### PR TITLE
Remove teams support in shipcat.conf

### PR DESCRIPTION
**Pull request by clux**
Last teams removed internally. Drops parsing of internal teams struct.

It also ports over `shipcat get resources` into more useful `shipcat top` flags.
In particular:

```
shipcat top --squads
shipcat top --tribes
shipcat top --squads --world -oyaml
```

etc.

### CODE REVIEW

**Comment by clux** [comment]
In the next PR we will reintroduce notificationMode as a per-service flag.

File: shipcat_cli/src/get.rs [hunk 1]
```diff
@@ -51,33 +51,6 @@ pub fn codeowners(conf: &Config) -> Result<Vec<String>> {
         let md = mf.metadata;
         let mut ghids = vec![];
         match conf.serviceOwnership {
```
Comment (janslow) [inline shipcat_cli/src/get.rs]: Can we get rid of this `match`?

File: shipcat_cli/src/lib.rs [hunk 2]
```diff
@@ -82,7 +82,7 @@ error_chain! {
 
 pub use shipcat_definitions::{Manifest, ConfigType};
 pub use shipcat_definitions::structs;
-pub use shipcat_definitions::config::{self, ConfigFallback, Config, Team};
+pub use shipcat_definitions::config::{self, ConfigFallback, Config};
```
Comment (janslow) [inline shipcat_cli/src/lib.rs]: Why is this exporting anything?

Comment (clux) [reply shipcat_cli/src/lib.rs]: for main.rs - which uses shipcat_cli's lib.

File: shipcat_cli/src/get.rs (hunk 1 above)
Comment (clux) [reply shipcat_cli/src/get.rs]: Wasn't sure if we were going to do a chapter/tribe for owners, so left the match in, but happy to kill it.

File: shipcat_definitions/src/config.rs [hunk 3]
```diff
@@ -185,10 +124,6 @@ pub struct Config {
     /// Gihub parameters
     pub github: GithubParameters,
 
-    /// Team definitions
-    #[serde(default)]
-    pub teams: Vec<Team>,
-
     /// Allowed labels
     #[serde(default)]
     pub allowedLabels: Vec<String>,
```
Comment (janslow) [inline shipcat_definitions/src/config.rs]: Can you get rid of `serviceOwnership` as well (or make it optional)?

File: shipcat_cli/src/get.rs (hunk 1 above)
Comment (janslow) [reply shipcat_cli/src/get.rs]: Surely that won't need a global `serviceOwnership` property though?

I guess it'd be a `chapter` or `tribe` property on the service metadata, which is mutually exclusive with `squad`

Comment (clux) [reply shipcat_cli/src/get.rs]: We will need the enum if we want to introduce the choice, and it doesn't feel like that's decided yet. It's easy enough to remove once it is though.

File: shipcat_definitions/src/config.rs (hunk 3 above)
Comment (clux) [reply shipcat_definitions/src/config.rs]: Ah, it is already optional because it implements default.

File: shipcat_cli/src/get.rs (hunk 1 above)
Comment (janslow) [reply shipcat_cli/src/get.rs]: Does it need a flag at all though? It's not a breaking change.

Comment (clux) [reply shipcat_cli/src/get.rs]: I suppose not, it's just annoying to have to potentially re-do it if we do make a decision.

Comment (janslow) [reply shipcat_cli/src/get.rs]: It's annoying to have `if (true) doThing()` too :P

Comment (clux) [reply shipcat_cli/src/get.rs]: Yeh. Alright, killing it.

**Pull request merged**